src/electron-main.ts

```diff
@@ -549,6 +560,51 @@ app.on("ready", async () => {
         global.mainWindow?.webContents.send("openDesktopCapturerSourcePicker");
         setDisplayMediaCallback(callback);
     });
+
+    session.defaultSession.webRequest.onBeforeRequest((req, callback) => {
+        // This handler emulates the element-web service worker, where URLs are rewritten late in the request
+        // for backwards compatibility. As authenticated media becomes more prevalent, this should be replaced
+        // by the app using authenticated URLs from the outset.
```

Comment:
Won't that mean if someone uses `Copy Image URL` and shares that link they'd be leaking their primary access token which is non-refreshing

Reply:
No, the copy function doesn't appear to call into this code at all.

Reply:
> As authenticated media becomes more prevalent, this should be replaced
        // by the app using authenticated URLs from the outset.

Yes but if the `img` `src` contains the token then copy-image-url would include that. Or is this just saying that this half of the rewrite logic would die but the headers/auth inserting would remain

Reply:
oh, unless you mean the future implied by the comment. We won't be using query string auth, so I'm not concerned there. It'll still be a service worker/electron hook which actually adds the authentication as a header. The URL would just be `/_matrix/client/v1/media` instead, eliminating the weird rewrite code in both places.